libs/payments/ui/src/lib/actions/recordEmitterEvent.ts

```diff
@@ -8,26 +8,6 @@ import { getAdditionalRequestArgs } from '../utils/getAdditionalRequestArgs';
 import { PaymentProvidersType } from '@fxa/payments/cart';
 import { PaymentsEmitterEventsKeysType } from '@fxa/payments/events';
 
-async function recordEmitterEventAction(
-  eventName: PaymentsEmitterEventsKeysType,
-  params: Record<string, string | string[]>,
-  searchParams: Record<string, string>
-): Promise<void>;
-
-async function recordEmitterEventAction(
-  eventName: 'checkoutFail',
-  params: Record<string, string | string[]>,
-  searchParams: Record<string, string>,
-  paymentProvider?: PaymentProvidersType
-): Promise<void>;
-
-async function recordEmitterEventAction(
-  eventName: 'checkoutSubmit' | 'checkoutSuccess',
-  params: Record<string, string | string[]>,
-  searchParams: Record<string, string>,
-  paymentProvider: PaymentProvidersType
-): Promise<void>;
-
```

Comment:
This change doesn't look related to the PR -- what's the intent here?

Reply:
The only substantial difference between them is whether or not the paymentProvider param is present or not. Since this PR causes us to no longer have guarantees that the paymentProvider field is populated at these stages, the type narrowing here is not used